mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp

```diff
@@ -1582,10 +1583,89 @@ template struct Conv2DRewritePattern<Conv2DOp>;
 template struct Conv2DRewritePattern<Conv2DBwdDataOp>;
 template struct Conv2DRewritePattern<Conv2DBwdWeightOp>;
 
+//===- MITPRewritePattern -------------------------------------------------===//
+//===-  ------------------------------------------------===//
+struct MITPRewritePattern : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  miopen::TransformOp makeTranspose(PatternRewriter &b, Value inp,
+                                    const AffineMapAttr &inMap,
+                                    const AffineMapAttr &outMap) const {
+    AffineMap inpIdxMap = inMap.getAffineMap();
+    AffineMap outpIdxMap = outMap.getAffineMap();
+    auto loc = inp.getLoc();
```

Comment:
Same deal - `Location` here, `ArrayRef<int64_t>` on the shape and so on ... standards upstream are to only use `auto` where the type is mentioned in the thing that generates the variable or it's a rather long and unwieldy type to write down.